fireworks/scripts/lpad_run.py

```diff
@@ -691,6 +692,115 @@ def maintain(args):
     lp.maintain(args.infinite, args.maintain_interval)
 
 
+def orphaned(args):
+    # get_fws
+    lp = get_lp(args)
+    if sum([bool(x) for x in [args.fw_id, args.name, args.state, args.query]]) > 1:
+        raise ValueError('Please specify exactly one of (fw_id, name, state, query)')
+    if sum([bool(x) for x in [args.fw_id, args.name, args.state, args.query]]) == 0:
+        args.query = '{}'
+        args.display_format = args.display_format if args.display_format else 'ids'
+    if sum([bool(x) for x in [args.fw_id, args.name, args.qid]]) > 1:
+        raise ValueError('Please specify exactly one of (fw_id, name, qid)')
+    else:
+        args.display_format = args.display_format if args.display_format else 'more'
+
+    if args.fw_id:
+        query = {'fw_id': {"$in": args.fw_id}}
+    elif args.name and not args.launches_mode:
+        query = {'name': args.name}
+    elif args.state:
+        query = {'state': args.state}
+    elif args.query:
+        query = ast.literal_eval(args.query)
+    else:
+        query = None
+
+    if args.sort:
+        sort = [(args.sort, ASCENDING)]
+    elif args.rsort:
+        sort = [(args.rsort, DESCENDING)]
+    else:
+        sort = None
+
+    if args.qid:
+        fw_ids = lp.get_fw_ids_from_reservation_id(args.qid)
+        if query:
+            query['fw_id'] = {"$in": fw_ids}
+            fw_ids = lp.get_fw_ids(query, sort, args.max, launches_mode=args.launches_mode)
+    else:
+        fw_ids = lp.get_fw_ids(query, sort, args.max,
+                               launches_mode=args.launches_mode)
```

Comment:
This whole section introduces a lot of code duplication from the ```get_fws``` function. Is it possible to create a third (helper) function with this code snippet that both ```get_fws``` and ```orphaned`` can use in order to avoid this duplication? e.g. ```get_fws_helper``` or something

Reply:
Sorry for not touching this for a long time, https://github.com/materialsproject/fireworks/pull/404/commits/72e88889fc09417dd33fcdd5fc719200e3ceac45 introduces such helper functions (split in two, get_fw_ids_helper, and get_fws_helper). I did not test the changes in that commit, relying on the CI here.